libethereum/Block.cpp

```diff
@@ -430,56 +448,76 @@ pair< TransactionReceipts, bool > Block::sync(
     return ret;
 }
 
-tuple< TransactionReceipts, unsigned > Block::syncEveryone(
-    BlockChain const& _bc, const Transactions& _transactions, uint64_t _timestamp, u256 _gasPrice,
-    Transactions* vecMissing  // it's non-null only for PARTIAL CATCHUP
-) {
+inline void Block::doPartialCatchupTestIfRequested( unsigned i ) {
+    static const char* FAIL_AT_TX_NUM = std::getenv( "TEST_FAIL_AT_TX_NUM" );
+    static int64_t transactionCount = 0;
+
+    if ( FAIL_AT_TX_NUM ) {
+        if ( transactionCount == std::stoi( FAIL_AT_TX_NUM ) ) {
+            // fail hard for test
+            cerror << "Test: crashing skaled on purpose after processing  " << i
```

Comment:
should we remove this before merging?

Reply:
No, it is a way to run partial catchup test, you need a way to crash in a middle of block processing